### PR TITLE
Update CI to run cargo make test fails out of the box

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Update CI to run cargo make test fails out of the box

### Issues Resolved
Follow up on https://github.com/opensearch-project/opensearch-rs/pull/117

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -8,6 +8,7 @@ STACK_VERSION = { value = "latest", condition = { env_not_set = ["STACK_VERSION"
 CARGO_MAKE_CARGO_PUBLISH_FLAGS = "--dry-run"
 # RUST_BACKTRACE is set to "full" in cargo make's builtin makefiles/stable.toml
 RUST_BACKTRACE = { value = "0", condition = { env_not_set = ["RUST_BACKTRACE"]}}
+SECURE_INTEGRATION = { value = "true", condition = { env_not_set = ["SECURE_INTEGRATION"] }}
```
Comment (reta) [inline Makefile.toml:11]: This will run using HTTPs by default, == `cargo make test`

File: Makefile.toml [hunk 2]
```diff
@@ -8,6 +8,7 @@ STACK_VERSION = { value = "latest", condition = { env_not_set = ["STACK_VERSION"
...
 OPENSEARCH_PROTOCOL_SUFFIX = { value = "s", condition = { env_true = ["SECURE_INTEGRATION"] }}
 OPENSEARCH_URL = { value = "http${OPENSEARCH_PROTOCOL_SUFFIX}://localhost:9200", condition = { env_not_set = ["OPENSEARCH_URL"] }}
```
Comment (dblock) [inline Makefile.toml:13]: I think if `SECURE_INTEGRATION=false`, don't you get `"http${OPENSEARCH_PROTOCOL_SUFFIX}://localhost:9200"` as `OPENSEARCH_URL`? The value of `OPENSEARCH_PROTOCOL_SUFFIX` is not blank, it's not set.

**Comment by dblock** [comment]
Add these to .gitignore

```
esnode-key.pem
esnode.pem
root-ca.pem
```

**Comment by dblock** [comment]
I am afraid `cargo make unittest` will fail with this. When I was doing serverless support I had lots of problems with docker and didn't want to deal with that suite of tests. I think keeping integration is probably the best way forward. WDYT?

**Comment by reta** [comment]
> Add these to .gitignore
> 
> ```
> esnode-key.pem
> esnode.pem
> root-ca.pem
> ```

I will add them to cleanup stage, right after container build up

**Comment by reta** [comment]
> I am afraid `cargo make unittest` will fail with this. When I was doing serverless support I had lots of problems with docker and didn't want to deal with that suite of tests. I think keeping integration is probably the best way forward. WDYT?

Ah I see now why you split them up, sure, I will ensure `cargo make unittest` works as expected, thanks!

File: Makefile.toml (hunk 2 above)
Comment (dblock) [reply Makefile.toml:13]: Didn't work for me :(

File: Makefile.toml [hunk 3]
```diff
@@ -158,7 +158,7 @@ run_task = "stop-opensearch"
 [tasks.run-unittest]
 private = true
 command = "cargo"
-args = ["test"]
+args = ["test", "--lib", "--bins"]
```
Comment (reta) [inline Makefile.toml:161]: @dblock the `test --lib --bins` runs only unit tests, so we should be good here

**Comment by reta** [comment]
`failed to get successful HTTP response from `[https://crates.io/api/v1/crates/anyhow/1.0.68/download`](https://crates.io/api/v1/crates/anyhow/1.0.68/download%60), got 503` :(

File: CHANGELOG.md [hunk 4]
```diff
@@ -26,7 +26,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ### Fixed
 - [BUG] cargo make test fails out of the box ([#117](https://github.com/opensearch-project/opensearch-rs/pull/117))
-
+- Update CI to run cargo make test fails out of the box ([#120](https://github.com/opensearch-project/opensearch-rs/pull/120))
```
Comment (dblock) [inline CHANGELOG.md:29]: Let's put an empty line after? ;)

File: Makefile.toml [hunk 5]
```diff
@@ -8,8 +8,8 @@ STACK_VERSION = { value = "latest", condition = { env_not_set = ["STACK_VERSION"
 CARGO_MAKE_CARGO_PUBLISH_FLAGS = "--dry-run"
 # RUST_BACKTRACE is set to "full" in cargo make's builtin makefiles/stable.toml
 RUST_BACKTRACE = { value = "0", condition = { env_not_set = ["RUST_BACKTRACE"]}}
-OPENSEARCH_PROTOCOL_SUFFIX = { value = "s", condition = { env_true = ["SECURE_INTEGRATION"] }}
-OPENSEARCH_URL = { value = "http${OPENSEARCH_PROTOCOL_SUFFIX}://localhost:9200", condition = { env_not_set = ["OPENSEARCH_URL"] }}
+OPENSEARCH_PROTOCOL = { source = "${SECURE_INTEGRATION}", default_value = "https", mapping = { "false" = "http", "true" = "https" }}
```
Comment (dblock) [inline Makefile.toml:11]: Nice.

File: opensearch/tests/client.rs [hunk 6]
```diff
@@ -27,9 +27,8 @@
  * Modifications Copyright OpenSearch Contributors. See
  * GitHub history for details.
  */
-#![cfg(feature = "_integration")]
 
-pub mod common;
+ pub mod common;
```
Comment (dblock) [inline opensearch/tests/client.rs:31]: Kill this space.

**Comment by reta** [comment]
@dblock huh, it seems like I have  it running BUT ...

```
failures:
    certificate_certificate_ca_validation
    full_certificate_ca_chain_validation
    full_certificate_ca_validation
```

Taking a look why

**Comment by reta** [comment]
:partying_face: @dblock the cert tests are now fixed, but we have a legit test failures:

```
failures:

---- deprecation_warning_headers stdout ----
thread 'deprecation_warning_headers' panicked at 'assertion failed: !warnings.is_empty()', opensearch/tests/client.rs:232:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    deprecation_warning_headers

test result: FAILED. 16 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 3.72s
```

Seems like we have not run `cargo make test` for a while ... I will take a look

**Pull request merged**